iib/workers/tasks/build.py

```diff
@@ -392,6 +392,21 @@ def _skopeo_inspect(*args, use_creds=False):
     return json.loads(_run_cmd(cmd, exc_msg=exc_msg))
 
 
+def _validate_index_image(resolved_prebuild_from_index, unresolved_from_index):
+    """
+    Validates if the index image has changed since IIB build started
+
+    :param str resolved_prebuild_from_index: resolved index image before starting the build
+    :param str unresolved_from_index: unresolved index image provided as API input
+    :return: a boolean indicating if the index image has changed since IIB build started
```

Comment:
I think it'd be more intuitive if this raised an `IIBError` exception if the from index image changed saying `The supplied from index image changed during the IIB request. Please resubmit the request.`. Then because you have `link_error=error_callback` when you call the Celery task, it will make the request as failed with the state reason as the exception.